discovery-provider/alembic/versions/436c10e54758_create_profile_challenges.py

```diff
@@ -0,0 +1,76 @@
+"""create profile challenges
+
+Revision ID: 436c10e54758
+Revises: c43d19f0f04a
+Create Date: 2021-06-16 18:47:23.425818
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '436c10e54758'
+down_revision = 'c43d19f0f04a'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    connection = op.get_bind()
+    connection.execute('''
+        -- There is a single broken user with two rows marked is_current; fix them first
+        update users 
+        set is_current = false
+        where user_id = 61019 and is_current = true and blocknumber = 20539912;
+
+        -- Create the challenge_profile_completion rows from
+        -- user & aggregate_user tables
+        insert into challenge_profile_completion (user_id, profile_description, profile_name, profile_picture, profile_cover_photo, follows, favorites, reposts)
+        select 
+            u.user_id,
+            u.bio is not null as profile_description,
+            true as profile_name, 
+            (u.profile_picture is not null or u.profile_picture_sizes is not null) as profile_picture,
+            (u.cover_photo is not null or u.cover_photo_sizes is not null) as profile_cover_photo,
+            au.following_count >= 5 as follows,
+            au.track_save_count >= 1 as favorites,
+            au.repost_count >= 1 as reposts
+        from users u 
+        join aggregate_user au on au.user_id = u.user_id 
+        where u.is_current = true;
+
+        -- Lastly, create the user_challenge rows from the newly created
+        -- challenge_profile_completion rows.
+        insert into user_challenges (challenge_id, user_id, specifier, is_complete, current_step_count)
+        select 
+            'profile-completion' as challenge_id,
+            c.user_id as user_id,
+            c.user_id as specifier,
+            (c.profile_description and
+            c.profile_name and
+            c.profile_picture and
+            c.profile_cover_photo and
+            c.follows and
+            c.favorites and
+            c.reposts
+            ) as is_complete,
+            (cast(c.profile_description as int) +
+            cast(c.profile_name as int) +
+            cast(c.profile_picture as int) +
+            cast(c.profile_cover_photo as int) +
+            cast(c.follows as int) +
+            cast(c.favorites as int) +
+            cast(c.reposts as int)
+            ) as current_step_count
+        from challenge_profile_completion as c;
+
+    ''')
+    # ### end Alembic commands ###
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    pass
```

Comment:
any reason to not clear the user_challenge rows for `profile-completion`?

Reply:
I was hesitant to add a downgrade here because we don't want to drop all the user_challenge rows, since new rows could be added to the table after the migration runs